.storybook/stories/DataFiltering/components/DashboardPagination.tsx

```diff
@@ -24,8 +24,37 @@ export const DashboardPagination = () => {
 			})}
 			itemsPerPage={pagination.perPage}
 			onItemsPerPageChange={(perPage) =>
-				setPagination({ ...pagination, perPage })
+				setPagination({
+					page: getValidPage(
+						pagination.page,
+						totalItems,
+						perPage,
+						pagination.perPage
+					),
+					perPage,
+				})
 			}
 		/>
 	);
 };
+
+const getValidPage = (
+	currentPage: number,
+	totalItems: number,
+	perPage: number,
+	prevPerPage: number
+) => {
+	// So the user doesn't lose their position in the nav order,
+	// we find the lowest displayed item number and ensure it is
+	// still on the page when we change items per page.
+	const lowestDisplayedItemNumber = Math.floor(
```

Comment:
Should we use `Index` instead of `Number`?

Reply:
I usually steer away from `index` when it's not zero-based to offer some differentiation between an index in an array. It's a loosely held ideal, though, so I don't mind swapping it over, if you think `lowestDisplayedItemIndex` is better?

Reply:
- [ ] Explore `lowestDisplayedItemIndex` and `lowestDisplayedItemPosition`